src/v/kafka/requests/metadata_request.cc

```diff
@@ -360,11 +361,19 @@ get_topic_metadata(request_context& ctx, metadata_request& request) {
     std::vector<ss::future<metadata_response::topic>> new_topics;
 
     for (auto& topic : *request.topics) {
+        auto mts = model::materialized_or_source_topic(topic);
         if (auto md = ctx.metadata_cache().get_topic_metadata(
-              model::topic_namespace_view(cluster::kafka_namespace, topic));
+              model::topic_namespace_view(cluster::kafka_namespace, mts));
             md) {
-            res.push_back(metadata_response::topic::make_from_topic_metadata(
-              std::move(*md)));
+            auto mdr_topic = metadata_response::topic::make_from_topic_metadata(
+              std::move(*md));
+            if (mts != topic) {
```

Comment:
maybe we can encapsulate this logic in `metadata_response::topic::make_from_topic_metadata` ?

Reply:
done